test/integration_test.py

```diff
@@ -411,19 +412,15 @@ def __check_manifest(self, file: IO[bytes], uuid_field_name: str) -> List[Mappin
 
     def _check_curl_manifest(self, _catalog: CatalogName, response: bytes):
         text = TextIOWrapper(BytesIO(response))
-        urls, names = [], []
-        for line in text.readlines():
-            if line.startswith('--') or line == '':
-                pass  # skip empty lines and config lines like '--compressed\n'
-            elif m := re.fullmatch(r'url="(.+)"\n', line):
-                urls.append(m.group(1))
-                self.assertEqual(len(urls), len(names) + 1, line)
-            elif m := re.fullmatch(r'output="(.+)"\n', line):
-                names.append(m.group(1))
-                self.assertEqual(len(urls), len(names), line)
-        self.assertEqual(len(urls), len(names), f'{urls}, {names}')
-        log.info(f'Manifest contains {len(urls)} files.')
-        self.assertGreater(len(urls), 0)
+        num_files = 0
+        lines = (line.rstrip() for line in text.readlines()
+                 if not line == '\n' and not line.startswith('--'))
+        for group in grouper(lines, 2):
```

Comment:
As a strong opponent of tuple indexing, I suggest something like

```suggestion
         for first, second in grouper(lines, 2):
```

Reply:
Went with different variable names as `first` is being imported from `more_itertools`.
```
/home/travis/build/DataBiosphere/azul/test/integration_test.py:418:13: F402 import 'first' from line 65 shadowed by loop variable
```